### ISSUE TITLE
Razor Comment Colorization doesn't colorize all lines

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#27447]
Expected:
```razor
@code {
    private int currentCount = 0;

    @* public class SomeAbstractClassImpl : AbstractClass
    {
        var dt = new DateTime();
    }
    
    public class SomethingDisposable : IDisposable
    {
        
    } *@
}
```

![image](https://user-images.githubusercontent.com/14852843/97907444-25696d00-1cfa-11eb-8c25-8b25a38aff19.png)

**Comment by ryanbrandenburg** [comment]
I think this would be fixed by https://github.com/dotnet/aspnetcore/issues/26975.

**Comment by NTaylorMullen** [comment]
Looks like C# is taking precedence:
![image](https://user-images.githubusercontent.com/2008729/108258250-a9fac280-7114-11eb-91ef-8b23df98b0f5.png)

**Comment by TanayParikh** [comment]
Not sure if this is the same issue, but ran into:

![Multiline comment](https://user-images.githubusercontent.com/14852843/116945740-4705cb80-ac2d-11eb-8004-e52b14959133.gif)

in a .cshtml file (within JS). Not sure if `@*` is meant to be supported within .cshtml/.js or not (if not, then I wouldn't expect <kbd>CTRL</kbd>+<kbd>/</kbd> to comment the block with `@* *@`

**Comment by NTaylorMullen** [comment]
oo definitely supported. What version of VS? I presume main?

**Comment by TanayParikh** [comment]
> What version of VS? I presume main?

`VSCode v1.23.11`

**Comment by TanayParikh** [comment]
Working as expected with `//` in VS with <kbd>CTRL</kbd>+<kbd>K</kbd>+<kbd>C</kbd>
![VSCommenting](https://user-images.githubusercontent.com/14852843/116947709-520f2a80-ac32-11eb-8eb6-527b79d37e55.gif)

**Comment by TanayParikh** [comment]
VS, JS within CSHTML is interesting...
![VSCommenting-CSHTML](https://user-images.githubusercontent.com/14852843/116948038-22145700-ac33-11eb-9fc3-7860966fa6da.gif)


vs. in an HTML page:
![VSCommenting](https://user-images.githubusercontent.com/14852843/116947973-f09b8b80-ac32-11eb-902e-a696a6d08b9b.gif)

Version [31222.87.main]

**Comment by TanayParikh** [comment]
Though note how the highlighted area changes after comment in the HTML. Is this expected? cc/ @ToddGrun

**Comment by ryanbrandenburg** [comment]
Good catches @TanayParikh.

**Comment by NTaylorMullen** [comment]
Ah @ryanbrandenburg the VS piece cshtml JS piece is a little unavoidable right because of TextMate?

**Comment by captainsafia** [comment]
I just ran into this while developing in VS Code. Seems to only happen inside the `@code` block. Commented out Razor is properly grayed out.

![image](https://user-images.githubusercontent.com/1857993/117081738-5bad9680-acf5-11eb-886e-76ca2625bbd0.png)

**Comment by TanayParikh** [comment]
@ryanbrandenburg is it possible your fix in https://github.com/dotnet/aspnetcore-tooling/pull/3381 isn't in the most recent VSCode release?

**Comment by ryanbrandenburg** [comment]
@TanayParikh yeah looks like that PR missed the deadline by a day or two, should be fixed in VSCode just by releasing the latest.

**Comment by ryanbrandenburg** [comment]
Let's include addressing https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1322800 in this.

**Comment by ryanbrandenburg** [comment]
![image](https://user-images.githubusercontent.com/14987221/123493750-43117c80-d5d2-11eb-9629-465552fc4b5e.png)
The above is achieved by removing some TextMate color info.

**Comment by ryanbrandenburg** [comment]
We'll wait for the fix from VS platform for this.

**Comment by ryanbrandenburg** [comment]
Now that the work on the VS platform side is on it's way in the work-item here is to revert all the commented out TM theme areas.
